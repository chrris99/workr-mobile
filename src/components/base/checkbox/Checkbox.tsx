import { colors } from "@/design-system/colors/colors";
import { Icon } from "@/design-system/icons/Icon";
import { spacing } from "@/design-system/spacing/spacing";
import { StyleSheet, View } from "react-native";

interface CheckboxProps {
  selected: boolean;
}

export const Checkbox = ({ selected }: CheckboxProps) => {
  const selectionStyle = selected ? styles.selected : styles.notSelected;

  return (
    <View style={[styles.container, selectionStyle]}>
      {selected && (
        <Icon name={"Check"} color={"white"} size="XS" strokeWidth={4} />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: spacing["spacing-4"],
    height: spacing["spacing-4"],
    borderRadius: spacing["spacing-2"],
    justifyContent: "center",
    alignItems: "center",
  },
  selected: {
    borderWidth: 1,
    borderColor: colors["primary-700"],
    backgroundColor: colors["primary-700"],
  },
  notSelected: {
    borderRadius: spacing["spacing-2"],
    borderWidth: 1,
    borderColor: colors["gray-300"],
  },
});
