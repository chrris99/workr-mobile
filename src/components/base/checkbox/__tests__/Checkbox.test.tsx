import { render } from "@testing-library/react-native";
import { Checkbox } from "../Checkbox";

describe("checkbox", () => {
  describe("in selected state", () => {
    it("should render correctly", () => {
      const component = render(<Checkbox selected={true} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe("in unselected state", () => {
    it("should render correctly", () => {
      const component = render(<Checkbox selected={false} />);
      expect(component).toMatchSnapshot();
    });
  });
});
