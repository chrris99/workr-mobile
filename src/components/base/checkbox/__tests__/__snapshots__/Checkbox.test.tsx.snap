// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox in selected state should render correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 8,
        "height": 16,
        "justifyContent": "center",
        "width": 16,
      },
      {
        "backgroundColor": "#6941C6",
        "borderColor": "#6941C6",
        "borderWidth": 1,
      },
    ]
  }
>
  <Text
    style={
      [
        undefined,
        {
          "fontFamily": "Poppins_400Regular",
          "fontSize": 14,
          "lineHeight": 20,
        },
        {
          "color": "#101828",
        },
      ]
    }
  >
    Check
  </Text>
</View>
`;

exports[`checkbox in unselected state should render correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 8,
        "height": 16,
        "justifyContent": "center",
        "width": 16,
      },
      {
        "borderColor": "#D0D5DD",
        "borderRadius": 8,
        "borderWidth": 1,
      },
    ]
  }
/>
`;
